### PR TITLE
Switch function based on object literals

### LINKED ISSUE #23: support object literals switch function
Create an snippet implementation for a object literals based switch

### PR DESCRIPTION
**Pull request by gustavoalfaro**
closes #23

### CODE REVIEW

**Comment by gaboesquivel** [comment]
I'm not sure, I prefer traditional switch for readability. This one is hard to read. 
It is also coupled to the ES5 function.  I prefer to keep function and switch separate.
If we added this I would prefer arrow function.

@capaj what do you think about this ?

**Comment by gustavoalfaro** [comment]
I think isn't hard to read instead not so familiar here's a couple of examples:
Short way:

``` javascript
  getFoodType = (name) => {
    return {
      Neapolitan: 'Pizza',
      Pork: 'Meat',
      Chicken: 'Meat',
    }[name] || 'No type match'
  }
```

Longest Way:

``` javascript
getFoodType = (name) => {
    const foodTypes = {
      Neapolitan: 'Pizza',
      Pork: 'Meat',
      Chicken: 'Meat',
      default: 'No type match'
    }
		
   return foodTypes[name] || foodTypes.default
  }
```

also using functions for more calculations:

``` javascript
getSnack = (type) => {
  isDrink = () => 'Drink'
  
  isFood = () => 'Food'
  
  var snacks = {
    coke: isDrink,
    pepsi: isDrink,
    cookies: isFood,
    crisps: isFood,
  }
  return snacks[type]()
}
```

**Comment by capaj** [comment]
@gaboesquivel I myself don't really like switches. I have been bitten to many times by forgotten `break` and such. I'd love this addition. Of course with the arrow function. regular function is way too long.
I think this one should be the shortest:
```javascript
getFoodType = (name) => ({
    Neapolitan: 'Pizza',
    Pork: 'Meat',
    Chicken: 'Meat',
  }[name] || 'No type match'
)
```

@gustavoalfaro could you modify it into this?

**Comment by gaboesquivel** [comment]
alright, @gustavoalfaro please update the pull request to use arrow function instead.  ✌(◕‿-)

**Comment by gustavoalfaro** [comment]
Update done ✔️ @gaboesquivel

File: README.md [hunk 1]
```diff
@@ -245,6 +245,14 @@ function* ${1:name}(${1:arguments}) {
 }
 ```
 
+#### `sf⇥` switch function based on object literals
+```js
+${1:name} = (${2:switching}) => ({
```
Comment (gaboesquivel) [inline README.md]: I think you are missing `const`

File: snippets/functions.cson [hunk 2]
```diff
@@ -64,3 +64,11 @@
     \t${0}
     }
     """
+  "switch function":
+    prefix: "sf"
+    body: """
+    ${1:name} = (${2:switching}) => ({
```
Comment (gaboesquivel) [inline snippets/functions.cson]: same here, missing `const`

**Comment by gustavoalfaro** [comment]
😄 typo error found and fixed

**Review by gaboesquivel — APPROVED** [review]

**Pull request merged**